render/render_test.go

```diff
@@ -18,73 +18,159 @@ const (
 name: my-app`
 )
 
-func TestRenderMissingValue(t *testing.T) {
-	configFiles := []composetypes.ConfigFile{
-		{
-			Config: map[string]interface{}{
-				"version": "3",
-				"services": map[string]interface{}{
-					"foo": map[string]interface{}{
-						"image": "${imageName}:${version}",
-					},
-				},
-			},
-		},
+func TestSubstituteBracedParams(t *testing.T) {
```

Comment:
I don't see a test with the syntax with default value: `${my.param:-mydefault}`. Is it supported by the regexps?

Reply:
Only a-z, 0-9, '-','_' and '.' are supported by regexp. Do you want me to add ':' to it?